packages/abi-gen/test/generated_test/test/abi_gen_dummy_test.ts

```diff
@@ -0,0 +1,149 @@
+import { BlockchainLifecycle, devConstants, web3Factory} from '@0x/dev-utils';
+import { Web3ProviderEngine } from '@0x/subproviders';
+import { RevertReason } from '@0x/types';
+import { BigNumber, providerUtils } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+
+import * as chai from 'chai';
+import * as chaiAsPromised from 'chai-as-promised';
+import * as ChaiBigNumber from 'chai-bignumber';
+import * as dirtyChai from 'dirty-chai';
+
+import { AbiGenDummyContract, artifacts, TestLibDummyContract } from '../src';
+
+const txDefaults = {
+    from: devConstants.TESTRPC_FIRST_ADDRESS,
+    gas: devConstants.GAS_LIMIT,
+};
+
+const provider: Web3ProviderEngine = web3Factory.getRpcProvider({ shouldUseInProcessGanache: true });
+const web3Wrapper = new Web3Wrapper(provider);
+
+chai.config.includeStack = true;
+chai.use(ChaiBigNumber());
+chai.use(dirtyChai);
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+describe('AbiGenDummy Contract', () => {
+    let abiGenDummy: AbiGenDummyContract;
+    before(async () => {
+        providerUtils.startProviderEngine(provider);
+        abiGenDummy = await AbiGenDummyContract.deployFrom0xArtifactAsync(artifacts.AbiGenDummy, provider, txDefaults);
+    });
+    after(async () => {
+        provider.stop();
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+    });
+    afterEach(async () => {
```

Comment:
I suggest doing the `blockchainLifecycle.startAsync()` and `...revertAsync()` in `before()` and `after()` rather than `...each()`.  No need to undeploy and redeploy the contract for every test case, since you're not mutating contract state with your test cases (right?).  Won't make THAT much of a difference, but our tests need every little speed-up they can get.